docs/src/main/sphinx/sql/table-functions.rst

```diff
@@ -0,0 +1,198 @@
+                new ConnectorTableFunctionHandle() {});
+    }
+
+The ``analyze()`` method returns an ``Analysis`` object, which comprises all
+the information required by the engine to analyze, plan and execute the table
+function invocation:
+
+- the returned row type, specified as an optional ``Descriptor``. It should be
+  passed if and only if the table function is declared with the
+  ``GENERIC_TABLE`` returned type.
+- dependencies between descriptor arguments and table arguments. Currently
+  descriptor and table arguments are not supported, so it should be
+  ``EMPTY_MAPPING``.
+- any information gathered during analysis that might be useful during planning
+  or execution, in the form of a ``ConnectorTableFunctionHandle``.
+  ``ConnectorTableFunctionHandle`` is a marker interface intended to carry
+  information throughout subsequent phases of query processing in a manner that
+  is opaque to the engine.
+
+Table function invocation
+-------------------------
+
+You invoke a table function in the ``FROM`` clause. Table function invocation
+syntax is similar to a scalar function call. There are two conventions of
+passing arguments to a table function:
+
+- arguments passed by name::
+
+    SELECT * FROM my_function("row_count" => 100, "column_count" => 1)
+
+In this convention, you can pass the arguments in arbitrary order. Arguments
+declared with default values can be skipped.
+
+- arguments passed positionally::
+
+    SELECT * FROM my_function(1, 100)
+
+In this convention, you must follow the order in which the arguments were
+declared. You can skip a suffix of the argument list, provided that all the
+skipped arguments are declared with default values.
+
+You cannot mix the argument conventions in one invocation.
+
+All arguments must be constant expressions, and they can be of any SQL type,
+which is compatible with the declared argument type.
+
+function and argument resolution
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The table function name is resolved case-insensitive, analogically to scalar
+function and table resolution in Trino. The arguments are resolved with regard
+to SQL identifier semantics, that is, case-sensitive, and with automatic
+uppercasing of unquoted names.
+
+Table function execution
+------------------------
+
+Currently, table functions are executed through pushdown to connector. The
+connector that provides a table function should support the
+``applyTableFunction()`` method. This method returns a ``ConnectorTableHandle``
+and a list of ``ColumnHandle`` s representing the table function result.
+
+.. code-block:: java
+
+    @Override
+    public Optional<TableFunctionApplicationResult<ConnectorTableHandle>> applyTableFunction(ConnectorSession session, ConnectorTableFunctionHandle handle)
+    {
+        // get the ConnectorTableHandle and a list of ColumnHandles representing the table function result
+
+        return Optional.of(new TableFunctionApplicationResult<>(tableHandle, columnHandles));
```

Comment:
Let's not copy code into docs to avoid going out of sync. Pointer to a  method (interface method name to implement) should suffice and then that method javadoc should describe the technical details

Reply:
In this case, I can indeed remove the pasted code, as it doesn't add value to the docs.
However, I'd like to leave the other code samples (the constructor and analyze()), even at the risk of going out of sync, because they provide a convenient way of explaining things. Anyway, I hope I won't miss the docs update if the SPI changes in the future.